packages/flutter/lib/src/widgets/semantics_debugger.dart

```diff
@@ -332,14 +332,14 @@ class _SemanticsDebuggerPainter extends CustomPainter {
     return childrenDepth + 1;
   }
 
-  void _paint(Canvas canvas, SemanticsNode node, int rank) {
+  void _paint(Canvas canvas, SemanticsNode node, int rank, int indexInParent) {
     canvas.save();
     if (node.transform != null) {
       canvas.transform(node.transform!.storage);
     }
     final Rect rect = node.rect;
     if (!rect.isEmpty) {
-      final Color lineColor = Color(0xFF000000 + math.Random(node.id).nextInt(0xFFFFFF));
+      final Color lineColor = Color(0xFF000000 + math.Random(_getColorSeed(rank, indexInParent)).nextInt(0xFFFFFF));
```

Comment:
So this is actually deterministic? Would it be possible to use the node's uid instead of logic in `_getColorSeed` to ensure the uniqueness (Is `math.Random` an injective function)?